### ISSUE TITLE
No result and no output

### ISSUE THREAD

**Issue by wiertz** [BernieO/calcardbackup#8]
I've followed the instructions, but running `sudo -u www-data ./calcardbackup /var/www/nextcloud/ -i -x` neither produces a directory with content nor any (error) message.

- Prerequisites are fullfilled
- Nextcloud 13.0.4 is up and running
- Server certificate is valid (not self-signed)
- users.txt file contains valid users and passwords
- users have personal and shared calendar data

Any suggestions what's wrong? Is there a way to make the script more verbose?

**Comment by BernieO** [comment]
I can't reproduce the issue.
Could you change line 22 (file calcardbackup)
from `set -euo pipefail`
to `set -euxo pipefail`
and run the script again and post the output?

**Comment by wiertz** [comment]
This is the output:
```
+ version='0.5.0 (09.06.2018), AGPL-3.0'
+ set_required_paths
++ pwd
+ working_dir=/root/calcardbackup
++ readlink -f ./calcardbackup
+ script_dir=
```

**Comment by BernieO** [comment]
Thank you.

The problem is that user `www-data` is not allowed to open the `/root` directory and thus is not able to resolve the (possibly relative) pathnames to absolute pathnames.

I don't think I can change this without losing versatility (e.g. calling the script from a different directory). I admit, I didn't have a proper think about that yet, though.

Meanwhile: could you try to use calcardbackup in a directory for which `www-data` is allowed to open all directorys in the path (change line 22 back to the original state before, otherwise you'll get quite a lot of output)?

**Comment by BernieO** [comment]
I decided to not change this behaviour due to versatility and backwards compatibility.

Version 0.5.1 prints a proper error message when this happens (absolute paths can't be read). I also updated the requirements in README.md (user running the script needs to be able to read full path to used files and to the script itself).

Closing this now - feel free to open it again, if you have any further questions.
